### PR TITLE
fix: missed enums and typings from #7290

### PR DESCRIPTION
**Pull request by IRONM00N**
**Please describe the changes this PR makes and why it should be merged:**
Some typings are out of date and a few enums are still being converted to strings. I also `{@link}`ed the enums in the JSDocs, if you want me to revert that I will.

**Status and versioning classification:**
- I know how to update typings and have done so, or typings don't need updating
- This PR changes the library's interface (methods or parameters added)
- This PR includes breaking changes (methods removed or renamed, parameters moved or removed)

### CODE REVIEW

File: packages/discord.js/src/managers/GuildScheduledEventManager.js [hunk 1]
```diff
@@ -44,24 +40,24 @@ class GuildScheduledEventManager extends CachedManager {
    * @property {string} name The name of the guild scheduled event
    * @property {DateResolvable} scheduledStartTime The time to schedule the event at
    * @property {DateResolvable} [scheduledEndTime] The time to end the event at
-   * <warn>This is required if `entityType` is `GuildScheduledEventEntityType.External`</warn>
+   * <warn>This is required if `entityType` is {@link GuildScheduledEventEntityType.External External}</warn>
```
Comment (ImRodry) [inline packages/discord.js/src/managers/GuildScheduledEventManager.js:43]: I’d say you don’t need the text, just leave the link itself

Comment (IRONM00N) [reply packages/discord.js/src/managers/GuildScheduledEventManager.js]: Okay, I changed it.

**Review by iCrawl — APPROVED** [review]

**Review by SpaceEEC — APPROVED** [review]

File: packages/discord.js/src/util/EnumResolvers.js [hunk 2]
```diff
@@ -231,6 +241,188 @@ class EnumResolvers extends null {
...
+   * @param {string|GuildScheduledEventEntityType} key The key to lookup
+   * @returns {GuildScheduledEventEntityType}
+   */
+  static resolveGuildScheduledEventEntityType(key) {
+    switch (key) {
+      case 'STAGE_INSTANCE':
+        return GuildScheduledEventEntityType.StageInstance;
+      case 'VOICE':
+        return GuildScheduledEventEntityType.Voice;
+      case 'External':
```
Comment (kyranet) [inline packages/discord.js/src/util/EnumResolvers.js:420]: Why is this PascalCase when the others are UPPER_SNAKE_CASE?

**Review by kyranet — CHANGES_REQUESTED** [review]

**Comment by suneettipirneni** [comment]
Just realized this also fixes a bug, can you go ahead and edit your PR description to add: "Closes 7336"

**Review by vladfrangu — APPROVED** [review]
For a future PR we should strictly type the strings users can input in the enum resolvers

**Pull request merged**